File: apps/web/src/interfaces/responses/GitStatusResponse.ts
export interface GitFileChange {
  status: string; // "M" | "A" | "D" | "R" | "?"
  path: string;
}

export interface GitStatusResponse {
  isInitialized: boolean;
  branch: string;
  unpushedCommits: number;
  modifiedFiles: number;
  isSynced: boolean;
  autoCommitEnabled: boolean;
  autoCommitIntervalSeconds: number;
  changedFiles: GitFileChange[];
}

File: apps/api/src/TerraformExplorer/Models/Responses/GitStatusResponse.cs
namespace TerraformExplorer.Models.Responses;

public class GitFileChange
{
    public string Status { get; set; } = ""; // M, A, D, R, ?
    public string Path { get; set; } = "";
}

public class GitStatusResponse
{
    public bool IsInitialized { get; set; }
    public string Branch { get; set; } = "";
    public int UnpushedCommits { get; set; }
    public int ModifiedFiles { get; set; }
    public bool IsSynced { get; set; }
    public bool AutoCommitEnabled { get; set; }
    public int AutoCommitIntervalSeconds { get; set; }
    public List<GitFileChange> ChangedFiles { get; set; } = [];
}
